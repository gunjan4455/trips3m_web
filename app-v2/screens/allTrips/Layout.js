import React, { Component } from 'react';
import { Helmet } from 'react-helmet';

import AllTrips from 'app-v2/modules/trip/allTrips';
import { rescue } from 'helpers/utils';
import { trackSegment } from 'actions/segmentEvents';

class Layout extends Component {

  constructor(props) {
    super(props);
  }

  trackSegmentEvent = (data = {}) => {
    rescue(() => {
      trackSegment({
        event: data.event,
        category: data.category || 'TravellerDashboard',
        section: data.section || '',
        object: data.object,
        cta: data.cta,
        misc_id_1: data.miscId1,
        misc_type_1: data.miscType1,
        misc_id_2: data.miscId2,
        misc_type_2: data.miscType2,
        misc_id_3: data.miscId3,
        misc_type_3: data.miscType3,
        misc_id_4: data.miscId4,
        misc_type_4: data.miscType4,
        misc_id_5: data.miscId5,
        misc_type_5: data.miscType5
      });
    });
  };


  render() {
    const AllTripsContainer = AllTrips.container;

    return (
      <div>
        <Helmet>
          <meta name="page_fullname" content="All Trips" />
          <meta name="page_name" content="All Trips" />
          <meta name="name" content="All Trips" />
          <meta charSet="utf-8" />
          <title>Traveler Dashboard - TravelTriangle</title>
          <link rel="canonical" href="https://traveltriangle.com/travellers/dashboard" />
        </Helmet>

        <AllTripsContainer trackSegment={this.trackSegmentEvent} />

      </div>
    );
  }
}

export default Layout;
