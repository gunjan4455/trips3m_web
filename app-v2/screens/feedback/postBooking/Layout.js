import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import Sensor from 'react-visibility-sensor';

import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { trackSegment } from 'actions/segmentEvents';
import common from 'app-v2/common';
import Header from 'modules/feedback/components/Header';
import Roster from 'modules/roster/index';
import PostBooking from 'modules/feedback/postBooking/index';
import { rescue } from 'helpers/utils';

class Layout extends Component {

  constructor(props) {
    super(props);
    this.isEventTracked = {};
  }

  componentDidMount() {
    /*    this.props.layout.components.map((component) => {
          this.isEventTracked[component.name] = false;
        });*/
  }

  UNSAFE_componentWillReceiveProps(props, nextProps) {
    /*if (props.currentTripId !== nextProps.currentTripId) {
      this.isEventTracked = {};
    }*/
  }

  trackSegmentEvent = (data = {}) => {
    rescue(() => {
      trackSegment({
        event: data.event,
        category: data.category || 'RLP',
        section: data.section || '',
        object: data.object,
        cta: data.cta,
        misc_id_1: data.miscId1,
        misc_type_1: data.miscType1,
        misc_id_2: data.miscId2,
        misc_type_2: data.miscType2,
        misc_id_3: data.miscId3,
        misc_type_3: data.miscType3,
        misc_id_4: data.miscId4,
        misc_type_4: data.miscType4,
        misc_id_5: data.miscId5,
        misc_type_5: data.miscType5
      });
    });
  };

  onSensorChange = (isVisible, config) => {
    if (isVisible) {
      if (!this.isEventTracked[config.name]) {
        this.isEventTracked[config.name] = true;

        this.trackSegmentEvent({
          event: ORGANISM_VIEWED,
          section: config.name,
          cta: ''
        });
      }
    }
  };

  render() {

    return (
      <div className="clearfix">

        <Helmet>
          <meta name="page_fullname" content="Quote Detail" />
          <meta name="page_name" content="Quote Detail" />
          <meta name="name" content="Quote Detail" />
        </Helmet>

        <Roster timeout={3000} />

        <div className="clearfix contentBackground sbc5">
          <div className="container">
            <div className="row">
              <Header text='Feedback' />
              <PostBooking />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Layout.propTypes = {};

export default Layout;
