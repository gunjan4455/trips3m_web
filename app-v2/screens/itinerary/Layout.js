import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import Sensor from 'react-visibility-sensor';

import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { trackSegment } from 'actions/segmentEvents';
import Itinerary from 'modules/quote/itinerary/detail/index';
import Roster from 'modules/roster';

class Layout extends Component {

  constructor(props) {
    super(props);
  }

  trackSegmentEvent = (data = {}) => {
    const { tripId, currentQuote, currentQuoteId } = this.props;

    if (!currentQuote) {
      return;
    }

    trackSegment({
      event: data.event,
      category: data.category || 'QDP',
      section: data.section || '',
      object: data.object || `Quote/${currentQuoteId}`,
      cta: data.cta,
      trip_id: tripId || (getCookie(TRIP_ID_COOKIE_KEY), 10) || 0,
      quote_id: currentQuoteId,
      invoice_id: currentQuote.invoiceDetails.id || 0,
      destination_id: currentQuote.destinationsList.length ? currentQuote.destinationsList[0].id : 0,
      agent_id: currentQuote.agent.id || 0,
      trip_status: currentQuote.tripStatus || '',
      quote_status: currentQuote.status || '',
      misc_id_1: data.miscId1,
      misc_type_1: data.miscType1,
      misc_id_2: data.miscId2,
      misc_type_2: data.miscType2,
      misc_id_3: data.miscId3,
      misc_type_3: data.miscType3,
      misc_id_4: data.miscId4,
      misc_type_4: data.miscType4,
      misc_id_5: data.miscId5,
      misc_type_5: data.miscType5
    });
  };

  onSensorChange = (isVisible) => {
    if (isVisible) {
      this.trackSegmentEvent({
        event: ORGANISM_VIEWED,
        section: 'Itinerary',
        cta: ''
      });
    }
  };

  render() {
    const ItineraryContainer = Itinerary.container;

    return (
      <div className="clearfix">

        <Helmet>
          <meta name="page_fullname" content="Quote Detail" />
          <meta name="page_name" content="Quote Detail" />
          <meta name="name" content="Quote Detail" />
        </Helmet>

        <Roster timeout={3000} />

        <div className="clearfix contentBackground sbc5">
          <div className="container">
            <div className="row pb48">
              <Sensor onChange={(isVisible) => this.onSensorChange(isVisible)}
                      partialVisibility='bottom'
                      delayedCall={true}>
                <ItineraryContainer trackSegment={(data) => this.trackSegmentEvent(data)} />
              </Sensor>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Layout.propTypes = {
  layout: PropTypes.object.isRequired,
  params: PropTypes.object.isRequired,
  quotes: PropTypes.object.isRequired,
  currentQuote: PropTypes.object.isRequired,
  currentQuoteId: PropTypes.number.isRequired,
  tripId: PropTypes.number.isRequired
};


export default Layout;
