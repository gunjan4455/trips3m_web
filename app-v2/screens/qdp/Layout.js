import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import Sensor from 'react-visibility-sensor';

import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { trackSegment } from 'actions/segmentEvents';
import { getCookie, setCookie } from 'helpers/FormDataFormatter';
import PriceMatchingPopup from 'modules/quote/cheaperDeal';
import common from 'app-v2/common';
import { getQuotesList, getDestination } from 'modules/quote/reducers/quoteList';
import Header from 'modules/quote/Header';
import QuoteDetailHeader from 'modules/quote/QuoteDetailHeader';
import QuoteTimer from 'modules/quote/QuoteTimer/QuoteTimer';
import QuoteStickyFooter from 'modules/quote/footer';
import Roster from 'modules/roster';
import ChatIcon from 'modules/quote/shared/ChatIcon';
import ThingsToCheck from './thingsToCheck/ThingsToCheck';
import GrowingMarket from './growingMarket/GrowingMarket';
import { checkIsTripStatusBeforeInvoiced } from 'helpers/utils';
import { checkIfQuoteActiveOrChosen } from 'modules/quote/reducers/quoteList';

class Layout extends Component {

  constructor(props) {
    super(props);
    this.isEventTracked = {};
    this.state = { pricePopup: false };
  }

  componentDidMount() {
    this.props.layout.components.map((component) => {
      this.isEventTracked[component.name] = false;
    });
  }

  UNSAFE_componentWillReceiveProps(props, nextProps) {
    if (props.currentQuoteId !== nextProps.currentQuoteId) {
      this.isEventTracked = {};
    }
  }

  trackSegmentEvent = (data = {}) => {
    const { tripId, currentQuote, currentQuoteId } = this.props;

    if (!currentQuote) {
      return;
    }

    trackSegment({
      event: data.event,
      category: data.category || 'QDP',
      section: data.section || '',
      object: data.object || `Quote/${currentQuoteId}`,
      cta: data.cta,
      trip_id: tripId || (getCookie(TRIP_ID_COOKIE_KEY), 10) || 0,
      quote_id: currentQuoteId,
      invoice_id: currentQuote.invoiceDetails.id || 0,
      destination_id: currentQuote.destinationsList.length ? currentQuote.destinationsList[0].id : 0,
      agent_id: currentQuote.agent.id || 0,
      trip_status: currentQuote.tripStatus || '',
      quote_status: currentQuote.status || '',
      // no_of_installments: currentQuote.installmentInfo && currentQuote.installmentInfo.installmentsReceived,
      // invoice_amount: currentQuote.invoiceDetails && currentQuote.invoiceDetails.price,
      // next_installment: currentQuote.installmentInfo && currentQuote.installmentInfo.nextScheduledDate,
      // invoice_curr: currentQuote.pricingInfo && currentQuote.pricingInfo.currency,
      misc_id_1: data.miscId1,
      misc_type_1: data.miscType1,
      misc_id_2: data.miscId2,
      misc_type_2: data.miscType2,
      misc_id_3: data.miscId3,
      misc_type_3: data.miscType3,
      misc_id_4: data.miscId4,
      misc_type_4: data.miscType4,
      misc_id_5: data.miscId5,
      misc_type_5: data.miscType5
    });
  };

  onSensorChange = (isVisible, config) => {
    if (isVisible) {
      if (!this.isEventTracked[config.name]) {
        this.isEventTracked[config.name] = true;
        if (config.name === 'Cabs') {
          const { currentQuoteId, quotes } = this.props;
          const currentQuote = quotes[currentQuoteId];
          if (checkIsTripStatusBeforeInvoiced(currentQuote.tripStatus)) {
            this.checkForPopup();
          }
        }
        this.trackSegmentEvent({
          event: ORGANISM_VIEWED,
          section: config.name,
          cta: ''
        });
      }
    }
  };

  checkForPopup = () => {
    let popupViewedTrips = getCookie('pmpop_trips');
    popupViewedTrips = popupViewedTrips ? JSON.parse(popupViewedTrips) : [];
    if (!popupViewedTrips.includes(this.props.tripId)) {
      popupViewedTrips.push(this.props.tripId);
      setCookie('pmpop_trips', JSON.stringify(popupViewedTrips));
      this.setState({ pricePopup: true });
    }
  }

  closePopup = () => {
    this.setState({ pricePopup: false });
  }

  render() {
    const { currentQuoteId, quotes, userDetails, quoteTimeRemaining } = this.props;
    const QuoteStickyFooterContainer = QuoteStickyFooter.container;
    const quotesList = getQuotesList(quotes);
    const currentQuote = quotes[currentQuoteId];
    const destination = getDestination(currentQuote);
    const allQuotesActiveOrChosen = quotesList.every(quote => checkIfQuoteActiveOrChosen(quote));

    return (
      <div className="clearfix">

        <Helmet>
          <meta name="page_fullname" content="Quote Detail" />
          <meta name="page_name" content="Quote Detail" />
          <meta name="name" content="Quote Detail" />
          <meta name="destination" content={destination} />
          <meta name="trip_status" content={currentQuote.tripStatus} />
          <meta name="quote_status" content={currentQuote.status} />
        </Helmet>

        <Roster timeout={3000} />

        <div className="clearfix contentBackground sbc5">
          <div className="container">
            <div className="row pb48">
              <Header
                tripId={this.props.tripId}
                currentQuoteId={currentQuoteId}
                userDetails={userDetails}
                trackSegment={(data) => this.trackSegmentEvent(data)}
              />
              {
                quoteTimeRemaining > 0 && allQuotesActiveOrChosen ?
                  <QuoteTimer
                    totalTimeInSeconds={quoteTimeRemaining}
                    trackSegment={(data) => this.trackSegmentEvent(data)}
                  /> : null
              }
              <QuoteDetailHeader
                quotesList={quotesList}
                currentQuoteId={this.props.currentQuoteId}
                tripId={this.props.tripId}
                location={this.props.location}
              />
              {
                this.props.layout.components.map((component, $index) => {
                  const ComponentsInstance = common[component.name];
                  return (
                    <Sensor key={$index}
                            onChange={(isVisible) => this.onSensorChange(isVisible, component)}
                            partialVisibility='bottom'
                            delayedCall={true}>
                      <div key={`comp_${$index}`}>
                        <ComponentsInstance config={component} destination={destination.split(',')[0]}
                                            trackSegment={(data) => this.trackSegmentEvent(data)} />
                      </div>
                    </Sensor>

                  );
                })
              }
              {/*end of quote*/}
              <div className="row row-">
                <div className="col-xs-12">
                  <p className="fw9 pfc4 text-center p24">End of Quote</p>
                </div>
              </div>
              {
                this.state.pricePopup ?
                  <PriceMatchingPopup
                    closePopup={this.closePopup}
                    trackSegment={(data) => this.trackSegmentEvent(data)}
                  /> : null
              }

              {/* faq*/}

              {/*to check*/}
              <ThingsToCheck />
              <div className="row row-">
                <div className="col-xs-12">
                  <p className="sfc8 pb24 pt24 pl15 pr15 ">We, at Travel Triangle make sure that the you enjoy the
                    inclusions promised by our agents.</p>
                </div>
              </div>

              {/*growing market*/}
              <GrowingMarket />
              <ChatIcon />
              <QuoteStickyFooterContainer trackSegment={(data) => this.trackSegmentEvent(data)} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Layout.propTypes = {
  layout: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  quotes: PropTypes.object.isRequired,
  quoteTimeRemaining: PropTypes.number,
  currentQuote: PropTypes.object.isRequired,
  currentQuoteId: PropTypes.number.isRequired,
  tripId: PropTypes.number.isRequired,
  userDetails: PropTypes.object.isRequired
};


export default Layout;
