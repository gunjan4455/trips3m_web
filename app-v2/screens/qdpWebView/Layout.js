import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import Sensor from 'react-visibility-sensor';

import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { trackSegment } from 'actions/segmentEvents';
import common from 'app-v2/common';
import Roster from 'modules/roster';
import { useQuery } from 'containers/Listing/parseFilters';

class Layout extends Component {

  constructor(props) {
    super(props);
    this.isEventTracked = {};
  }

  trackSegmentEvent = (data = {}) => {
    const { tripId, currentQuote, currentQuoteId } = this.props;

    if (!currentQuote) {
      return;
    }

    trackSegment({
      event: data.event,
      category: data.category || 'QDP',
      section: data.section || '',
      object: data.object || `Quote/${currentQuoteId}`,
      cta: data.cta,
      trip_id: tripId || (getCookie(TRIP_ID_COOKIE_KEY), 10) || 0,
      quote_id: currentQuoteId,
      invoice_id: currentQuote.invoiceDetails.id || 0,
      destination_id: currentQuote.destinationsList.length ? currentQuote.destinationsList[0].id : 0,
      agent_id: currentQuote.agent.id || 0,
      trip_status: currentQuote.tripStatus || '',
      quote_status: currentQuote.status || '',
      misc_id_1: data.miscId1,
      misc_type_1: data.miscType1,
      misc_id_2: data.miscId2,
      misc_type_2: data.miscType2,
      misc_id_3: data.miscId3,
      misc_type_3: data.miscType3,
      misc_id_4: data.miscId4,
      misc_type_4: data.miscType4,
      misc_id_5: data.miscId5,
      misc_type_5: data.miscType5
    });
  };

  onSensorChange = (isVisible, config) => {
    if (isVisible) {
      if (!this.isEventTracked[config.name]) {
        this.isEventTracked[config.name] = true;

        this.trackSegmentEvent({
          event: ORGANISM_VIEWED,
          section: config.name,
          cta: ''
        });
      }
    }
  };

  render() {

    const { search } = this.props.location;
    const query = useQuery(search);
    const ComponentsInstance = common[query.component];

    return (
      <div className="clearfix">

        <Helmet>
          <meta name="page_fullname" content="Quote Detail" />
          <meta name="page_name" content="Quote Detail" />
          <meta name="name" content="Quote Detail" />
        </Helmet>

        <Roster timeout={3000} />

        <div className="clearfix contentBackground sbc5">
          <div className="container">
            <div className="row pb48">
              {
                  <ComponentsInstance trackSegment={(data) => this.trackSegmentEvent(data)} />
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Layout.propTypes = {
  location: PropTypes.object.isRequired,
  layout: PropTypes.object.isRequired,
  quotes: PropTypes.object.isRequired,
  currentQuote: PropTypes.object.isRequired,
  currentQuoteId: PropTypes.number.isRequired,
  tripId: PropTypes.number.isRequired
};


export default Layout;
