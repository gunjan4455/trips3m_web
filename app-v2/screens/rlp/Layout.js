import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import Sensor from 'react-visibility-sensor';

import { ORGANISM_VIEWED } from 'actions/segmentEvents';
import { trackSegment } from 'actions/segmentEvents';
import common from 'app-v2/common';
import NavBar from 'modules/rlp/NavBar';
import Header from 'modules/rlp/header';
import Roster from 'modules/roster';
import { rescue } from 'helpers/utils';
import AppDownloadBanner from 'modules/rlp/appDownloadBanner';

class Layout extends Component {

  constructor(props) {
    super(props);
    this.isEventTracked = {};
    this.quotesApiInterval = null;
    this.counter = 0;
  }

  resetCounter = () => this.counter = 0;

  checkQuoteAtInterval = (props) => {
    const { layout, fetchQuotesData, match: { params }} = props;

    this.resetCounter();
    clearInterval(this.quotesApiInterval);

    if (layout.stage === 'LEAD_CREATED') {
      this.quotesApiInterval = setInterval(() => {
        if (this.counter >= 4) {
          return clearInterval(this.quotesApiInterval);
        }
        this.counter++;
        fetchQuotesData({ tripId: params.requestedTripId });
      }, 10000);
    }
  };

  componentDidMount() {
    const { layout } = this.props;
    layout.components.map((component) => {
      this.isEventTracked[component.name] = false;
    });

    this.checkQuoteAtInterval(this.props);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (this.props.currentTripId !== nextProps.currentTripId) {
      this.isEventTracked = {};

      this.checkQuoteAtInterval(nextProps);
    }

    const prevQuotes = this.props.quotes && Object.keys(this.props.quotes) || [];
    const currentQuotes = nextProps.quotes && Object.keys(nextProps.quotes) || [];
    if (prevQuotes.length === 0 && currentQuotes.length && this.props.layout.stage === 'LEAD_CREATED') {
      this.props.showRoster({
        allowTimeOut: false,
        message: ['Dear traveler, refresh the page to view your first quote. ',
          <a onClick={() => window.location.reload()}><u>View Quote</u></a>]
      });
      clearInterval(this.quotesApiInterval);
    }
  }

  componentWillUnmount(){
    clearInterval(this.quotesApiInterval);
  }

  trackSegmentEvent = (data = {}) => {
    rescue(() => {
      trackSegment({
        event: data.event,
        category: data.category || 'RLP',
        section: data.section || '',
        object: data.object,
        cta: data.cta,
        misc_id_1: data.miscId1,
        misc_type_1: data.miscType1,
        misc_id_2: data.miscId2,
        misc_type_2: data.miscType2,
        misc_id_3: data.miscId3,
        misc_type_3: data.miscType3,
        misc_id_4: data.miscId4,
        misc_type_4: data.miscType4,
        misc_id_5: data.miscId5,
        misc_type_5: data.miscType5
      });
    });
  };

  onSensorChange = (isVisible, config) => {
    if (isVisible) {
      if (!this.isEventTracked[config.name]) {
        this.isEventTracked[config.name] = true;

        this.trackSegmentEvent({
          event: ORGANISM_VIEWED,
          section: config.name,
          cta: ''
        });
      }
    }
  };

  render() {
    const { trips, currentTripId, layout } = this.props;
    const currentTrip = trips.byId[currentTripId];

    return (
      <div className="clearfix">

        <Helmet>
          <meta name="page_fullname" content="Request Landing" />
          <meta name="page_name" content="Request Landing" />
          <meta name="name" content="Request Landing" />
          <meta name="destination" content={currentTrip.firstDestinationName} />
          <meta name="trip_stage" content={layout.stage} />
          <meta name="trip_status" content={currentTrip.status} />
        </Helmet>

        <Roster timeout={3000} />
        <div className="clearfix contentBackground sbc5">
          <div className="at_appDownloadBannerSection">
            <Sensor onChange={(isVisible) => this.onSensorChange(isVisible, { name: 'App Banner' })}
                    partialVisibility='bottom'
                    delayedCall={true}>
              <AppDownloadBanner trackSegment={this.trackSegmentEvent} />
            </Sensor>
          </div>
          <div className="container">
            <div className="row pb48">
              <NavBar currentTripId={currentTripId} />
              <Header currentTrip={currentTrip} trackSegment={this.trackSegmentEvent} />
              {
                this.props.layout.components.map((component, $index) => {
                  const ComponentsInstance = common[component.name];
                  return (
                    <Sensor key={$index}
                            onChange={(isVisible) => this.onSensorChange(isVisible, component)}
                            partialVisibility='bottom'
                            delayedCall={true}>
                      <ComponentsInstance config={component}
                                          trackSegment={(data) => this.trackSegmentEvent(data)} />
                    </Sensor>
                  );
                })
              }
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Layout.propTypes = {
  layout: PropTypes.object.isRequired,
  match: PropTypes.object.isRequired,
  trips: PropTypes.object.isRequired,
  currentTripId: PropTypes.number.isRequired,
  showRoster: PropTypes.func,
  fetchQuotesData: PropTypes.func,
  quotes: PropTypes.object
};


export default Layout;
