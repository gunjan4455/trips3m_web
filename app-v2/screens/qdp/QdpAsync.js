import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Layout from './Layout';
import { useQuery } from 'containers/Listing/parseFilters';

export default class QdpAsync extends Component {
  constructor(props) {
    super(props);

    if (!props.quotes.hasOwnProperty(props.currentQuoteId)) {
      throw new Error('404');
    }
  }

  componentDidMount() {
    this.updateQuoteStatus(this.props);
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (this.props.currentQuoteId !== nextProps.currentQuoteId) {
      this.updateQuoteStatus(nextProps);
    }
  }

  updateQuoteStatus = (props) => {
    const query = useQuery(props.location.search);
    const source = query.from_tt_mailer || '0';

    props.updateQuoteStats({ tripId: props.tripId, quoteId: props.currentQuoteId, source });
  };

  render() {
    return (
      <Layout {...this.props} />
    );
  }
}

QdpAsync.propTypes = {
  layout: PropTypes.object.isRequired,
  location: PropTypes.object.isRequired,
  quotes: PropTypes.object.isRequired,
  quoteTimeRemaining: PropTypes.number.isRequired,
  currentQuote: PropTypes.object.isRequired,
  currentQuoteId: PropTypes.number.isRequired,
  tripId: PropTypes.number.isRequired,
  updateQuoteStats: PropTypes.func.isRequired,
  pushState: PropTypes.func.isRequired
};
