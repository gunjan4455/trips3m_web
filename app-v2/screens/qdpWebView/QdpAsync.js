import React, { Component } from 'react';
import PropTypes from 'prop-types';

import Layout from './Layout';

export default class QdpAsync extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Layout {...this.props} />
    );
  }
}

QdpAsync.propTypes = {
  location: PropTypes.object.isRequired,
  layout: PropTypes.object.isRequired,
  quotes: PropTypes.object.isRequired,
  currentQuote: PropTypes.object.isRequired,
  currentQuoteId: PropTypes.number.isRequired,
  tripId: PropTypes.number.isRequired,
};
