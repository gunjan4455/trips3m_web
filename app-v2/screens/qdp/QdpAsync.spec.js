import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import QdpAsync from './QdpAsync';
import Layout from './Layout';

configure({ adapter: new Adapter() });

const props = {
  layout: {
    'meta_info': {
      'destination': 'Andaman',
      'trip_status': 'Active'
    },
    'name': 'RLP-IR-11-07-2018',
    'stage': 'INVOICE_RECIEVED',
    'module_name': 'rlp',
    'components': [{
      'id': 1,
      'name': 'educateTraveller',
      'order': 1,
      'row_no': 1,
      'block_no': 1,
      'variant': 'FULL',
      'total_block_in_row': 1
    }, {
      'id': 2,
      'name': 'aboutDestination',
      'order': 2,
      'row_no': 2,
      'block_no': 1,
      'variant': 'FULL',
      'total_block_in_row': 1,
      'lookupParent': true
    }],
    'actions': [],
    'version': 1
  },
  params: {},
  location: { query: {} },
  updateQuoteStats: jest.fn,
  quotes: {
    12345: {
      id: 12345
    }
  },
  currentQuoteId: 12345
};

describe('RLP page', () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<QdpAsync {...props} />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Layout1', () => {
    expect(wrapper.find(Layout)).toHaveLength(1);
  });
});


