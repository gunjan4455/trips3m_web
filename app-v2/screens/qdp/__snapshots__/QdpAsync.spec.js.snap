// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RLP page should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QdpAsync
    currentQuoteId={12345}
    layout={
      Object {
        "actions": Array [],
        "components": Array [
          Object {
            "block_no": 1,
            "id": 1,
            "name": "educateTraveller",
            "order": 1,
            "row_no": 1,
            "total_block_in_row": 1,
            "variant": "FULL",
          },
          Object {
            "block_no": 1,
            "id": 2,
            "lookupParent": true,
            "name": "aboutDestination",
            "order": 2,
            "row_no": 2,
            "total_block_in_row": 1,
            "variant": "FULL",
          },
        ],
        "meta_info": Object {
          "destination": "Andaman",
          "trip_status": "Active",
        },
        "module_name": "rlp",
        "name": "RLP-IR-11-07-2018",
        "stage": "INVOICE_RECIEVED",
        "version": 1,
      }
    }
    location={
      Object {
        "query": Object {},
      }
    }
    params={Object {}}
    quotes={
      Object {
        "12345": Object {
          "id": 12345,
        },
      }
    }
    updateQuoteStats={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "currentQuoteId": 12345,
      "layout": Object {
        "actions": Array [],
        "components": Array [
          Object {
            "block_no": 1,
            "id": 1,
            "name": "educateTraveller",
            "order": 1,
            "row_no": 1,
            "total_block_in_row": 1,
            "variant": "FULL",
          },
          Object {
            "block_no": 1,
            "id": 2,
            "lookupParent": true,
            "name": "aboutDestination",
            "order": 2,
            "row_no": 2,
            "total_block_in_row": 1,
            "variant": "FULL",
          },
        ],
        "meta_info": Object {
          "destination": "Andaman",
          "trip_status": "Active",
        },
        "module_name": "rlp",
        "name": "RLP-IR-11-07-2018",
        "stage": "INVOICE_RECIEVED",
        "version": 1,
      },
      "location": Object {
        "query": Object {},
      },
      "params": Object {},
      "quotes": Object {
        "12345": Object {
          "id": 12345,
        },
      },
      "updateQuoteStats": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "currentQuoteId": 12345,
        "layout": Object {
          "actions": Array [],
          "components": Array [
            Object {
              "block_no": 1,
              "id": 1,
              "name": "educateTraveller",
              "order": 1,
              "row_no": 1,
              "total_block_in_row": 1,
              "variant": "FULL",
            },
            Object {
              "block_no": 1,
              "id": 2,
              "lookupParent": true,
              "name": "aboutDestination",
              "order": 2,
              "row_no": 2,
              "total_block_in_row": 1,
              "variant": "FULL",
            },
          ],
          "meta_info": Object {
            "destination": "Andaman",
            "trip_status": "Active",
          },
          "module_name": "rlp",
          "name": "RLP-IR-11-07-2018",
          "stage": "INVOICE_RECIEVED",
          "version": 1,
        },
        "location": Object {
          "query": Object {},
        },
        "params": Object {},
        "quotes": Object {
          "12345": Object {
            "id": 12345,
          },
        },
        "updateQuoteStats": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
